odc/stats/_cli_save_tasks.py

```diff
@@ -153,9 +159,18 @@ def on_message(msg):
     def gqa_predicate(ds):
         return ds.metadata.gqa_iterative_mean_xy <= gqa
 
+    def collection_category_predicate(ds):
+        if ds.type in ["ls5_sr", "ls7_sr", "ls8_sr", "ls9_sr"]:
+            return ds.metadata.collection_category == usgs_collection_category
+        else:
+            return True
```

Comment:
This implementation is weird to me.  Why isn't it just: 

`return ds.metadata.collection_category == usgs_collection_category`

Reply:
Because not all products have `collection_category`, and so it filters out all the Sentinel-2 datasets, for example.

Our use case is to be able to run the command:

```
odc-stats save-tasks --grid africa_30 --temporal-range 2021-12--P1M --usgs-collection-category=T2 ls8_sr-ls9_sr-s2_l2a
```